### PR TITLE
refactor(test): separate protocol and dialect in the RDBMS integration tests

### PR DESCRIPTION
**Pull request by eitsupi**
Part of #2813, continued from #2804

### CODE REVIEW

**Comment by eitsupi** [comment]
@max-sixty Sorry, but I could not figure out how to implement `import_csv` without using match as you wrote here https://github.com/PRQL/prql/pull/2804#discussion_r1226086555.

File: prql-compiler/tests/integration/main.rs [hunk 1]
```diff
@@ -74,21 +89,124 @@ impl IntegrationTest for Dialect {
...
+                    ),
+                })
+            }
             _ => None,
         }
     }
 }
 
+impl SetUpData for DbConnection {
+    fn import_csv(&mut self, csv_name: &str, runtime: &Runtime) {
```
Comment (max-sixty) [inline prql-compiler/tests/integration/main.rs]: I think this _does_ probably require the Big Match Statement — the alternative would be a Struct per Dialect, which I think would be too much structure given we already have a Struct per Protocol.

---

I think it's not worth changing (even if my thought is correct, which may not be), but one question — could this be part of `IntegrationTest` trait, and have the methods receive a `protocol`? Generally implementing on the smallest unit keeps it simple. `modify_sql` doesn't even take a `protocol` arg.

File: prql-compiler/tests/integration/main.rs [hunk 2]
```diff
@@ -119,16 +237,16 @@ fn test_fmt_examples() {
 fn test_rdbms() {
     let runtime = &*RUNTIME;
 
-    let mut connections: Vec<(Dialect, Box<dyn DbConnection>)> = Dialect::iter()
+    let mut connections: Vec<DbConnection> = Dialect::iter()
```
Comment (max-sixty) [inline prql-compiler/tests/integration/main.rs:263]: This is nicer...

File: prql-compiler/tests/integration/main.rs [hunk 3]
```diff
@@ -184,14 +303,15 @@ fn test_rdbms() {
     })
 }
 
-fn setup_connection(con: &mut dyn DbConnection, runtime: &Runtime) {
+fn setup_connection(con: &mut DbConnection, runtime: &Runtime) {
```
Comment (max-sixty) [inline prql-compiler/tests/integration/main.rs]: I guess this could be part of `DbConnection` (but this is me being very refine-y now, no need to change)

**Review by max-sixty — COMMENTED** [review]
Nice!

File: prql-compiler/tests/integration/main.rs (hunk 1 above)
Comment (eitsupi) [reply prql-compiler/tests/integration/main.rs]: That makes sense, thanks.
I updated that.

**Comment by max-sixty** [comment]
Looking good!

**Comment by eitsupi** [comment]
Thank you for your review. I think it is much better now.

**Pull request merged**